e2e-tests/ldap-tls/run

```diff
@@ -0,0 +1,176 @@
+     "db" : "admin"
+   },
+   {
+     "role" : "dbAdminAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "clusterMonitor",
+     "db" : "admin"
+   },
+   {
+     "role" : "readWriteAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "restore",
+     "db" : "admin"
+   },
+   {
+     "role" : "backup",
+     "db" : "admin"
+   }
+ ]})' \
+		"userAdmin:userAdmin123456@$cluster-rs0.$namespace"
+
+	run_mongo 'db.runCommand({connectionStatus:1})' "percona:password@$cluster-rs0.$namespace" "" "" "--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'"
+
+	roles_size=$(run_mongo 'JSON.stringify(db.runCommand({connectionStatus:1}))' \
+		"percona:password@$cluster-rs0.$namespace" "" "" \
+		"--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'" \
+		| grep -E -v "Percona Server for MongoDB|connecting to:|Implicit session:|versions do not match|Error saving history file:|bye" \
+		| jq '.authInfo.authenticatedUserRoles | length')
+
+	if [ -z "$roles_size" ]; then
+		echo "Got problem while getting roles size"
+		exit 1
+	fi
+
+	if [ "$roles_size" -eq 0 ]; then
+		echo "No roles found"
+		exit 1
+	fi
+
+	kubectl_bin delete psmdb $cluster
+	kubectl_bin delete pvc --all
+}
+
+test_sharded_openldap() {
+	cluster="some-name-sharded"
+
+	kubectl_bin create secret generic $cluster-mongos --from-literal=mongos.conf="$(yq "$test_dir/conf/mongos.conf" \
+		| yq ".security.ldap.servers=\"openldap.$namespace.svc.cluster.local:1636\"")"
+
+	kubectl_bin create secret generic $cluster-cfg-mongod --from-literal=mongod.conf="$(yq "$test_dir/conf/mongod.conf" \
+		| yq ".security.ldap.servers=\"openldap.$namespace.svc.cluster.local:1636\"")"
+
+	desc "create first PSMDB cluster $cluster"
+
+	apply_cluster "$test_dir/conf/$cluster.yml"
+
+	desc 'check if all 3 Pods started'
+	wait_for_running $cluster-rs0 3
+	wait_for_running $cluster-cfg 3 "false"
+	wait_for_running $cluster-mongos 3
+
+	run_mongos \
+		'db.getSiblingDB("admin").createRole(
+{
+ role: "cn=admin,ou=perconadba,dc=ldap,dc=local",
+ privileges: [],
+ roles : [
+   {
+     "role" : "readAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "dbAdminAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "clusterMonitor",
+     "db" : "admin"
+   },
+   {
+     "role" : "readWriteAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "restore",
+     "db" : "admin"
+   },
+   {
+     "role" : "backup",
+     "db" : "admin"
+   }
+ ]})' \
+		"userAdmin:userAdmin123456@$cluster-mongos.$namespace"
+
+	roles_size=$(run_mongos 'JSON.stringify(db.runCommand({connectionStatus:1}))' \
+		"percona:password@$cluster-mongos.$namespace" "" "" \
+		"--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'" \
+		| grep -E -v "Percona Server for MongoDB|connecting to:|Implicit session:|versions do not match|Error saving history file:|bye" \
+		| jq '.authInfo.authenticatedUserRoles | length')
+
+	if [ -z "$roles_size" ]; then
```

Comment:
it would be better if we checked for some string response and not just number of roles

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1480/commits/6e770f75cc0402167333b367a07563bcfba6b2de